grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy

```diff
@@ -1155,7 +1155,7 @@ class ShipmentService {
         }
 
         Holders.grailsApplication.mainContext.publishEvent(new ShipmentStatusTransitionEvent(shipmentInstance, ShipmentStatusCode.SHIPPED))
```

Comment:
the only thing coming to my mind for now is that the event class is not Transactional, thus it doesn't commit the change (you call `this.shipmentStatusCode = shipmentStatusCode` and it should be persisted when the session flushes, but only if you have transaction open). 
You can try this:
- marking `ShipmentStatusTransitionEvent` as Transactional